db/index_changes.go

```diff
@@ -402,9 +403,12 @@ func (db *Database) vectorChangesFeed(channel string, options ChangesOptions) (<
 		if options.Since.TriggeredByClock != nil {
 			for i := 0; i < len(log); i++ {
 				logEntry := log[i]
-				// If sequence is less than the backfillTo clock sequence for its vbucket, send as backfill (i.e. with triggered by)
-				isBackfill := logEntry.Sequence <= options.Since.TriggeredByClock.GetSequence(logEntry.VbNo)
 
+				// If vb.seq for the entry is earlier than the vb.seq that triggered this channel's backfill, send as backfill.
+				isBackfill := false
+				if logEntry.VbNo < options.Since.TriggeredByVbNo || (logEntry.VbNo == options.Since.TriggeredByVbNo && logEntry.Sequence < options.Since.TriggeredBy) {
```

Comment:
This is a really complicated compound conditional, can it get reworked into something that's a little easier to grok?

maybe something like:

```
backfillNeeded := isBackFillNeeded(param1, param2...)

// If vb.seq for the entry is earlier than the vb.seq that triggered this channel's backfill,
// then a backfill is needed
func isBackFillNeeded(param1, param2, ...) boolean {

}

```